test/Altinn.App.Core.Tests/Internal/LocaltestValidationTests.cs

```diff
@@ -0,0 +1,414 @@
+
+        var result = await service.Results.FirstAsync();
+        Assert.NotNull(result);
+        Assert.IsType<VersionResult.InvalidVersionResponse>(result);
+
+        Assert.False(lifetime.ApplicationStopping.IsCancellationRequested);
+    }
+
+    [Fact]
+    public async Task Test_Timeout()
+    {
+        await using var fixture = Fixture.Create();
+
+        var expectedVersion = 1;
+        var delay = TimeSpan.FromSeconds(6);
+
+        var server = fixture.Server;
+        server
+            .Given(Request.Create().WithPath(Fixture.ApiPath).UsingGet())
+            .RespondWith(
+                Response
+                    .Create()
+                    .WithStatusCode(200)
+                    .WithHeader("Content-Type", "text/plain")
+                    .WithBody($"{expectedVersion}")
+                    .WithDelay(delay)
+            );
+
+        var service = fixture.Validator;
+        var lifetime = fixture.App.Services.GetRequiredService<IHostApplicationLifetime>();
+        await service.StartAsync(lifetime.ApplicationStopping);
+        await Task.Delay(10);
+        fixture.TimeProvider.Advance(delay);
+
+        var result = await service.Results.FirstAsync();
+        Assert.NotNull(result);
+        Assert.IsType<VersionResult.Timeout>(result);
+
+        Assert.False(lifetime.ApplicationStopping.IsCancellationRequested);
+    }
+
+    [Fact]
+    public async Task Test_App_Shutdown()
+    {
+        await using var fixture = Fixture.Create();
+
+        var expectedVersion = 1;
+        var delay = TimeSpan.FromSeconds(6);
+
+        var server = fixture.Server;
+        server
+            .Given(Request.Create().WithPath(Fixture.ApiPath).UsingGet())
+            .RespondWith(
+                Response
+                    .Create()
+                    .WithStatusCode(200)
+                    .WithHeader("Content-Type", "text/plain")
+                    .WithBody($"{expectedVersion}")
+                    .WithDelay(delay)
+            );
+
+        var service = fixture.Validator;
+        var lifetime = fixture.App.Services.GetRequiredService<IHostApplicationLifetime>();
+        await service.StartAsync(lifetime.ApplicationStopping);
+        await Task.Delay(10);
+        fixture.TimeProvider.Advance(delay.Subtract(TimeSpan.FromSeconds(4)));
+        lifetime.StopApplication();
+
+        var result = await service.Results.FirstAsync();
+        Assert.NotNull(result);
+        Assert.IsType<VersionResult.AppShuttingDown>(result);
+
+        Assert.True(lifetime.ApplicationStopping.IsCancellationRequested);
+    }
+
+    [Fact]
+    public async Task Test_Dns_Failure()
+    {
+        await using var fixture = Fixture.Create(registerCustomAppServices: services =>
+            services.Configure<PlatformSettings>(settings =>
+                settings.ApiStorageEndpoint = ReplaceHost(
+                    settings.ApiStorageEndpoint,
+                    "provoke-dns-fail.local.altinn.cloud"
```

Comment:
Is this dependent on external dns and the test will fail when someone adds a wildcard *.local.altinn.cloud registration?

Reply:
yes! Could mock out dns resolution by resolving the host in a handler or something, but I wanted to leave the `HttpClient` mostly intact so that I get all the real exceptions and behavior. I changed the address to `provoke-dns-fail.local.not-altinn-at-all.cloud` instead. If someone decides to register that I will pay the price of changing the test 😄